internal/plugintest/working_dir_json_test.go

```diff
@@ -0,0 +1,69 @@
+package plugintest_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+// TestJSONConfig verifies that TestStep.Config can contain JSON.
+// This test also proves that when changing the HCL and JSON formats back and
+// forth, the framework deletes the previous configuration file.
+func TestJSONConfig(t *testing.T) {
+	providerFactories := map[string]func() (*schema.Provider, error){
+		"tst": func() (*schema.Provider, error) { return tstProvider(), nil },
+	}
+	resource.Test(t, resource.TestCase{
+		IsUnitTest:        true,
+		ProviderFactories: providerFactories,
+		Steps: []resource.TestStep{{
+			Config: `{"resource":{"tst_t":{"r1":{"s":"x1"}}}}`,
+			Check:  resource.TestCheckResourceAttr("tst_t.r1", "s", "x1"),
+		}, {
+			Config: `resource "tst_t" "r1" { s = "x2" }`,
+			Check:  resource.TestCheckResourceAttr("tst_t.r1", "s", "x2"),
+		}, {
+			Config: `{"resource":{"tst_t":{"r1":{"s":"x3"}}}}`,
+			Check:  resource.TestCheckResourceAttr("tst_t.r1", "s", "x3"),
+		}},
+	})
+}
+
+func tstProvider() *schema.Provider {
+	return &schema.Provider{
+		ResourcesMap: map[string]*schema.Resource{
+			"tst_t": &schema.Resource{
```

Comment:
Hopefully should appease the linter.

```suggestion
			"tst_t": {
```